source/connection-string/tests/valid-options.yml

```
@@ -15,3 +15,12 @@ tests:
             db: "admin"
         options:
             authmechanism: "MONGODB-CR"
+    -
+        description: "Missing delimiting slash between hosts and options"
+        uri: "mongodb://example.com?tls=true"
+        valid: true
+        warning: false
+        hosts: ~
```

This should also assert that the hostname is parsed correctly:

```suggestion
        hosts:
            -
                type: "hostname"
                host: "example.com"
                port: ~
```